### PR TITLE
MERC-5889:  SBRV Tests for Verifier's SetConfig

### PR DESCRIPTION
**Pull request by dav009**
[MERC-5889](https://smartcontract-it.atlassian.net/browse/MERC-5889)

### Resolves Dependencies
- https://github.com/smartcontractkit/chainlink/pull/13813

## What ?

- Added Tests for `DestinationVerifier.SetConfig`
- Added a couple extra `get` methods to verifier contract so we can easily check expected internal state

   
   
## Notes

> forge test --match-path "src/v0.8/llo-feeds/v0.4.0/test/verifier/*"

```
Suite result: ok. 12 passed; 0 failed; 0 skipped; finished in 13.61ms (19.62ms CPU time)
```   

[MERC-5889]: https://smartcontract-it.atlassian.net/browse/MERC-5889?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: contracts/src/v0.8/llo-feeds/v0.4.0/test/verifier/VerifierSetConfigTest.t.sol [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+        vm.expectRevert(abi.encodeWithSelector(DestinationVerifier.DONConfigDoesNotExist.selector));
+        s_verifier.setConfigActive(dummyDONConfigID, true);
+    }
+
+    function test_setConfigWithAddressesAndWeightsAreSetCorrectly() public {
+        Signer[] memory signers = _getSigners(MAX_ORACLES);
+        address[] memory signerAddrs = _getSignerAddresses(signers);
+        Common.AddressAndWeight[] memory weights = new Common.AddressAndWeight[](1);
+        weights[0] = Common.AddressAndWeight(signers[0].signerAddress, ONE_PERCENT * 100);
+        s_verifier.setConfig(signerAddrs, FAULT_TOLERANCE, weights);
```
Comment (dav009) [inline contracts/src/v0.8/llo-feeds/v0.4.0/test/verifier/VerifierSetConfigTest.t.sol]: todo: check side effect here

File: contracts/src/v0.8/llo-feeds/v0.4.0/DestinationVerifier.sol [hunk 2]
```diff
@@ -387,5 +388,13 @@ contract DestinationVerifier is IDestinationVerifier, ConfirmedOwner, TypeAndVer
     function getFeeManager() external view override returns (address) {
         return address(s_feeManager);
     }
+
+    function getDONConfigs() public view returns (DONConfig[] memory) {
```
Comment (Fletch153) [inline contracts/src/v0.8/llo-feeds/v0.4.0/DestinationVerifier.sol]: We don't want to expose any configuration details that could lead to parasitic usage. By exposing this, users could get the DONConfigs and verify without paying the feee

File: contracts/src/v0.8/llo-feeds/v0.4.0/DestinationVerifier.sol [hunk 3]
```diff
@@ -387,5 +388,13 @@ contract DestinationVerifier is IDestinationVerifier, ConfirmedOwner, TypeAndVer
...
+        return s_DONConfigs;
+    }
+
+     function getSignerConfigByAddressAndDONConfigId(bytes32 _id) public view returns (bool) {
```
Comment (Fletch153) [inline contracts/src/v0.8/llo-feeds/v0.4.0/DestinationVerifier.sol]: Same as above, these should all be private with no getters

**Comment by dav009** [comment]
@Fletch153  addressed comments in this PR:
- removed getter functions
- removed tests checking internal state
 - these will be addressed via calls to verify in another PR

**Review by Fletch153 — APPROVED** [review]

**Review by Fletch153 — APPROVED** [review]

**Pull request merged**